General structure in Geometry group (#36)
asam-ev/OpenMATERIAL: content/geometry/general-structure/General_structure.adoc

= General structure

== Dimensions
The unit used for object dimensions are meters.

== Coordinate systems
The local coordinate system is a right-handed coordinate system according to ISO 8855 with the axes pointing to the following directions. For a non-rotated coordinate system the following applies:

* Forward matches x-axis
* Left matches y-axis
* Up matches z-axis

> The coordinate system is in line with ASAM OpenDRIVE, ASAM OpenSCENARIO, ASAM OSI.

Author: ```suggestion
The coordinate system is in line with ASAM OpenDRIVE, ASAM OpenSCENARIO, and ASAM OSI.
```

////
Origins of the coordinate frames for specific object classes are under
discussion.
////

== Naming conventions
The following naming conventions apply to material files:

> * Use prefix in the 3D model file name to show that it is according to standard

Author: ```suggestion
* Use prefixes in the 3D model file name to show that it is up to standard
```

discussion.

== Naming conventions
The following naming conventions apply to ASAM OpenMATERIAL files:

* The name of the 3D model file shall have the prefix `omg_` to indicate that the file
complies with the ASAM OpenMATERIAL standard.
* The 3D model file and the related 3D asset file shall have the same base name.

*Example*
If the name of the 3D model file is `omg_my-model.gltf`, then the
3D asset file has either the name `omg_my-model.xoma` (for material assets) or
`omg_my-model.xomg`(for material geometries).

NOTE: Putting more information in file name to be discussed for specific object classes.

== 3D asset file
The 3D asset file is used for mapping ASAM OpenMATERIAL objects based on node, mesh,
or material names. It is stored in JSON format.

The 3D asset file provides metadata and material mapping to the geometry. The extension of
a 3D asset file indicates the content type provided:

* `xoma` for material assets. Example: `omg_my-model.xoma`.
* `xomg` for material geometries. Example: `omg_my-model.xomg`.

NOTE: Miro board mentions that the extensions `xoma` and `xomg` are "different
from material files". Clarify which terms to use for

> different files (open material geometry file = 3D model file?).

Reviewer: xoma and xomg were just two suggestions for the OpenMATERIAL asset file. I would suggest using `xoma` for now. This is also what I used in the examples. The material parameters will have a different file extension. In the preliminary examples I used `xomp` for "open material properties"

= General structure

== Dimensions
The unit used for object dimensions is meters.

== Coordinate systems
The local coordinate system is a right-handed coordinate system according to

> ISO 8855 with the axes pointing to the directions as follows. For a non-rotated

Reviewer 2: Rephrased: "The local coordinate system is a right-handed coordinate system according to
ISO 8855 with the axes pointing to the directions mentioned below. For a non-rotated
coordinate system, the following applies:"

coordinate system, the following applies:

* Forward matches x-axis
* Left matches y-axis
* Up matches z-axis

////
TODO: Add images similar to OpenDRIVE? See
https://publications.pages.asam.net/standards/ASAM_OpenDRIVE/ASAM_OpenDRIVE_Specification/latest/specification/08_coordinate_systems/08_04_local_coordinate_system.html
////

The coordinate system is in line with the standards ASAM OpenDRIVE,
ASAM OpenSCENARIO, and ASAM OSI.

////
Origins of the coordinate frames for specific object classes are under
discussion.
////

== Naming conventions
The following naming conventions apply to ASAM OpenMATERIAL files:

* The name of the 3D model file shall have the prefix `omg_` to indicate that the file
complies with the ASAM OpenMATERIAL standard.

> * The 3D model file and the related asset file shall have the same base name.

Author: ```suggestion
* The 3D model file and the related 3D asset file shall have the same base name.
```

*Example*
If the name of the 3D model file is `omg_my-model.gltf`, then the
asset file has name `omg_my-model.xoma` (for material assets) or
`omg_my-model.xomg`(for material geometries).

////
Putting more information in file name to be discussed for specific object classes.
////

> == Asset file

Author: ```suggestion
== 3D asset file
```

> The asset file is used for mapping ASAM OpenMATERIAL objects based on node, mesh,

Author: ```suggestion
The 3D asset file is used for mapping ASAM OpenMATERIAL objects based on node, mesh,
```

or material names. It is stored in JSON format.

> The asset file either provides material assets or material geometries. The extension of

Author: ```suggestion
The 3D asset file provides either material assets or material geometries. The extension of
```

> The 3D asset file provides either material assets or material geometries. The extension of

Reviewer 2: Sentence is rephrased.

> an asset file indicates the content type provided:

Author: ```suggestion
a 3D asset file indicates the content type provided:
```

> * `xoma` for material assets. Example: `omg_my-model.xoma`.

Reviewer 2: Yes, there is an empty line needed before a list in AsciiDoc. I will correct this.

> * `xoma` for material assets. Example: `omg_my-model.xoma`.

Reviewer: I will rewrite this section.

* `xomg` for material geometries. Example: `omg_my-model.xomg`.

////
Miro board mentions that the extensions `xoma` and `xomg` are "different
from material files". Clarify which terms to use for
different files (open material geometry file = 3D model file?).
////

> Each asset file contains the following information:

Author: ```suggestion
Each 3D asset file contains the following information:
```

* Metadata
* Material mapping (with possibility to easily change assignment)
* Optional link to another asset file to facilitate instancing (same asset file for different 3D models)
* Semantic information (labels) for sub-meshes ?
* Description of coordinates, pivot points and so on instead of hierarchy in 3D model file ?
* Assigning data streams (for example, wheel rotation) to certain sub-meshes ?

////
TODO: Add short descriptions and references
////

> 

Author: ```suggestion
Suggestion: Add section for material parameter files analogue to asset file 
```

> Suggestion: Add section for material parameter files analogue to asset file 

Reviewer 2: Deleted the sentence.
